File: arry/229. 求众数 II/java/Majority Element II.java
class Solution {
    /**
     * 利用哈希Map存储每个元素出现总数
     */
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> hash = new HashMap<>();
        List<Integer> elements = new ArrayList();
        int count = Math.floorDiv(nums.length, 3);
        for (int num : nums) {
            int c = hash.getOrDefault(num, 0) + 1;
            hash.put(num, c);
            if (count < c && !elements.contains(num)) {
                elements.add(num);
            }
        }
        return elements;
    }
}

class Solution2 {
    public List<Integer> majorityElement(int[] nums) {

    }
}

File: arry/229. 求众数 II/python/Majority Element II.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        # 指定大小
        result = []
        num = len(nums)/3
        ret = {}
        for i in nums:
            # 计算元素在数组中的个数
            if i in ret:
                ret[i] += 1
            else:
                ret[i] = 1
        
        for i in ret:
            if ret[i] > num:
                result.append(i)

        return result